src/tools/compiletest/src/runtest.rs

```diff
@@ -1958,23 +1958,16 @@ impl<'test> TestCx<'test> {
         let mut filecheck = Command::new(self.config.llvm_filecheck.as_ref().unwrap());
         filecheck.arg("--input-file").arg(output).arg(&self.testpaths.file);
 
-        // FIXME: Consider making some of these prefix flags opt-in per test,
-        // via `filecheck-flags` or by adding new header directives.
-
         // Because we use custom prefixes, we also have to register the default prefix.
         filecheck.arg("--check-prefix=CHECK");
 
-        // Some tests use the current revision name as a check prefix.
+        // Tests are allowed to use a revision name as a check prefix.
         if let Some(rev) = self.revision {
             filecheck.arg("--check-prefix").arg(rev);
         }
 
-        // Some tests also expect either the MSVC or NONMSVC prefix to be defined.
-        let msvc_or_not = if self.config.target.contains("msvc") { "MSVC" } else { "NONMSVC" };
-        filecheck.arg("--check-prefix").arg(msvc_or_not);
-
-        // The filecheck tool normally fails if a prefix is defined but not used.
-        // However, we define several prefixes globally for all tests.
+        // The filecheck tool normally fails if a prefix is defined but not used. However, sometimes
+        // revisions are used to specify *compiletest* directives which are not FileCheck concerns.
```

Comment:
Nitpick: If we're revising these comments, I think they should acknowledge that auto-registering revision names as check prefix is a bit sketchy, and that having to pass `--allow-unused-prefix` is an unfortunate side-effect of not knowing whether the test author actually wanted revision-specific check prefixes or not.

Reply:
The main reason I haven't tried to fix this yet is that there are quite a few tests affected, and in tests that use both CHECK and revision-specific prefixes, there's a risk of accidentally only checking the CHECK ones and silently ignoring all the revision ones.

Reply:
True. I'll think about this a bit more.

Reply:
> The main reason I haven't tried to fix this yet is that there are quite a few tests affected, and in tests that use both CHECK and revision-specific prefixes, there's a risk of accidentally only checking the CHECK ones and silently ignoring all the revision ones.

Don't we register the revisions via

```rs
if let Some(rev) = self.revision {
    filecheck.arg("--check-prefix").arg(rev);
}
```

?

Reply:
I added comments back + further clarified the current situation.

Reply:
Opened #134510 to track this and backlinked from comments.